### DISCUSSION TITLE
How to solve keymap collisions between Helix and Zellij?

### DISCUSSION THREAD

**Discussion by MarkuBu** [helix-editor/helix#8537] [Q&A]
I would like to use Helix in a Zellij session, but there are many collisions in the keyboard mapping. Zellij uses the same key combinations as Ctrl+s or some other Ctrl and Alt combinations.

This makes it difficult to use Helix within Zellij without having to do a lot of remapping. Are there any options other than remapping or not using Helix within a Zellij session?

**Comment by AngeloChecked** [comment]
I've resolved this issue by simply disabling Zellij using Ctrl+G. I've become quite accustomed to doing it, although initially, it didn't feel comfortable. I'm very eager to hear if there's a more effective approach.

**Reply by mydoghasfleas** [reply]
Oh, I did not realize that is what the "Lock" feature in Zellij was. I learned something, thanks! 😁

But it seems like quite a reasonable solution if you don't need the colliding shortcuts so often.

It's certainly a better solution than prefixing every single terminal multiplexer command with some combination, like on GNU screen.

**Comment by Dangerise** [comment]
+1 looking forward to a better solution

**Comment by DekuD2** [comment]
At last I arrived at a solution that doesn't hinder my workflow—make zellij work well even when it's locked. That way I can just stay in the locked mode 99% of the time. I added the following keybinds (to `~/.config/zellij/config.kdl`) to let me move around when zellij is locked, and then I just stay in the locked mode unless I need to change my tab/pane setup.

```
keybinds {
    shared {
        bind "Alt h" "Alt Left" { MoveFocusOrTab "Left"; }
        bind "Alt l" "Alt Right" { MoveFocusOrTab "Right"; }
        bind "Alt j" "Alt Down" { MoveFocus "Down"; }
        bind "Alt k" "Alt Up" { MoveFocus "Up"; }
        bind "Alt m" { ToggleFloatingPanes; }
    }
...
}
```

**Reply by Dangerise** [reply]
Thanks, it helped a lot

**Reply by nffdiogosilva** [reply]
But I'm assuming you then have different bindings to navigate between views within Helix. That's currently my main issue. Ideally I would be able to seamlessly navigate between them (Helix/Zellij), using the same keybindings.

**Comment by cbrake** [comment]
`Ctrl-b` (switch Zellij to tmux mode) is the main conflict I'm seeing right now. Disabled that with:

```
keybinds {
	shared_except "tmux" "locked" {
	//  bind "Ctrl b" { SwitchToMode "Tmux"; }
	    unbind "Ctrl b"
	}
}
```

**Comment by luccahuguet** [comment]
hey guys, this is the set of remaps i currently implement in my project, yazelix: https://github.com/luccahuguet/zellij#keybinding-remaps

you can either copy the relevant config (from yazelix.kdl) to your own zellij layout, or manually add the remaps to your zellij/config.toml or even use my project directly

**Reply by cbrake** [reply]
neat project -- thanks for sharing.

**Comment by ondrej-ivanko** [comment]
Zellij added configuration which allows updating the keymaping which lead to key conflicts. Now you can assign different key modifiers to commands which are used the most. Also Ctrl+g is an option. See: https://zellij.dev/documentation/rebinding-keys
